### DISCUSSION TITLE
How much performance will be reduced when using official plugins？

### DISCUSSION THREAD

**Discussion by iarno** [Kong/kong#7448] [Help] [answered]

**Comment by locao** [comment, marked as answer]
It depends on a lot of factors, as which plugins you are planning to use or the size of the requests/responses. The best approach is to simulate the environment you expect, run some load tests, and compare the results. 

The simple fact of adding a plugin is almost irrelevant to the general performance. As a rule of thumb, you can expect that any plugins that inspect the request/response body or make external calls will have a bigger impact.
